tradesman/model_creation/zoning/zone_building.py

```diff
@@ -18,29 +19,44 @@ def zone_builder(project, hexbin_size: int, max_zone_pop: int, min_zone_pop: int
          *min_zone_pop*(:obj:`int`): min population within a zone. Defaults to 500
          *save_hexbins*(:obj:`bool`): saves hexbins with population. Defaults to False
     """
+
+    print("\n begin", strftime("%Y-%m-%d %H:%M:%S", gmtime()))
     sql = "SELECT division_name, level, Hex(ST_AsBinary(geometry)) as geometry FROM political_subdivisions;"
     subdivisions = gpd.GeoDataFrame.from_postgis(sql, project.conn, geom_col="geometry", crs=4326)
     subdivisions = subdivisions[subdivisions.level == subdivisions.level.max()]
 
+    sql = "SELECT division_name, level, Hex(ST_AsBinary(geometry)) as geometry FROM political_subdivisions where level = -1;"
+    model_area = gpd.GeoDataFrame.from_postgis(sql, project.conn, geom_col="geometry", crs=4326)
+    if model_area.shape[0] > 0:
+        subdivisions = subdivisions[subdivisions.intersects(model_area.geometry.unary_union)]
+
```

Comment:
I think this code block is unnecessary. When we import the political subdivisions into the model, we only import those which intersect our model area. For instance: Coquimbo a Chilean state. We import the provinces belonging to the state, and the cities belonging to each province. The subdivisions we use to build the zones are the cities of each state (in line 26 of this code), so we don't have to check where the subdivisions intersect with the model area.

Reply:
I don't think so, as my local copy here had subdivisions for the whole country. Maybe it only happens when we do things in pieces?

Reply:
Geez. This is definitely unexpected. It isn't supposed to get subdivisions for the whole country, just the model area. I'll check this later.